### PR TITLE
Add Llama3.2-90B-Vision model

### PR DESCRIPTION
**Pull request by gwangTT**
### Ticket
#19622 

### Problem description
Customers are interested in running `Llama3.2-90B-Vision` model on TT hardware. `Llama3.2-11B-Vision` is already supported and we know there were going to be aspects of the code that do not scale to a much larger model such as the `90B`:
- Model weight loading is challenging because the checkpoints are sharded across 8 files (unlike `11B`)
- The smallest the machine to run `90B` is T3K and also `bfloat8_b` weights becomes a requirement (`11B` uses `bfloat16`)
- Adding such a large model to CI requires certain special treatment

Furthermore, it is desirable to arrive at code that natively provides a unified support for both `90B` and `11B` visions models. For example, avoid `if is_90B: ... else: ...` as much as possible.

### What's changed
This solves the problems as such:
- Model weight loading
  - Unified checks on whether a tensor is sharded across checkpoint files in `is_param_replicated_across_shards` function in `models/tt_transformers/tt/load_checkpoints.py`
  - Added `test_load_checkpoints.py` to check and make sure that the unified approach supports all llama models including the `90B`
  - Added `models/tt_transformers/scripts/repack_weights_90b.py` to reduce debugging time and to reduce CI test time for vision-model-only tests
- Problems with smaller dtype for model weights
  - Block floats requires `TILE_LAYOUT`, which created some shape mismatch problems for `90B`; this is fixed through a `ttnn.reshape(<tensor>, <logical_shape>, <padded_shape>)`
  - Needed to apply the same mixed-dtype optimization the same as `70B`
- Adding `90B` to CI
  - First hurdle is jumped through coordination with core-sw-infra team in order to generate weight cache files on mounted drives to CI machines; see [this issue](https://github.com/tenstorrent/tt-metal/issues/19622) for details.
  - various test cases are added to the following pipelines: `t3000-demo-tests`, `t3000-unit-tests`, `t3000-perplexity-tests`, and `t3000-frequent-tests`
  - fixed two bugs caught by the above tests: 1) TILE_LAYOUT does not support `ttnn.concat(dim=-2)` but I introduced it by mistake to `class_embedding`; 2) need to use smaller dtype for reference models in the tests or the model won't fit in memory.
  - limited the `n_layers` to `1` for the test in the `frequent-tests` group and also use saved reference tensor for one of the test case in that group as it still takes too long to run with `n_layers=1` on `pipeline-functional`.
- [x] To do before merging: add entry to README.md about newly supported model: [a456da1](https://github.com/tenstorrent/tt-metal/pull/20069/commits/a456da1d129d094305b3358de8139858e31b25ef)

### Checklist
- [x] All post commit CI passes: https://github.com/tenstorrent/tt-metal/actions/runs/14714861782
  - on  [b23a0cc](https://github.com/tenstorrent/tt-metal/pull/20069/commits/b23a0cccdacc4285bedbe7257d41ff70f58565c1)
- [x] T3K pipelines for the new model: https://github.com/tenstorrent/tt-metal/actions/runs/14714878233
  - on  [b23a0cc](https://github.com/tenstorrent/tt-metal/pull/20069/commits/b23a0cccdacc4285bedbe7257d41ff70f58565c1)
  - all passed except for a known unrelated failed test case, with team working on it here: https://tenstorrent.slack.com/archives/C08H57QF1RV/p1745945004221989?thread_ts=1745944244.428759&cid=C08H57QF1RV
- [x] Single test to make sure no regression introduced to existing models on single card: https://github.com/tenstorrent/tt-metal/actions/runs/14714927616
  - on  [b23a0cc](https://github.com/tenstorrent/tt-metal/pull/20069/commits/b23a0cccdacc4285bedbe7257d41ff70f58565c1)
  - single failure with test timing out during successful runs, due to a know, unrelated-to-this-PR issue, with team working on it here: https://tenstorrent.slack.com/archives/C05GRJC4J4A/p1746027062631929

### CODE REVIEW

File: models/tt_transformers/tests/multimodal/test_llama_class_embedding.py [hunk 1]
```diff
@@ -87,18 +87,18 @@ def test_class_embedding_inference(
...
     model_args = ModelArgs(mesh_device)
-    state_dict = torch.load(model_args.consolidated_weights_path, map_location=torch.device("cpu"))
+    state_dict = model_args.load_state_dict()
     first_layer_prefix = "vision_model.vision_encoder."
     partial_state_dict = {
         k[len(first_layer_prefix) :]: v for k, v in state_dict.items() if (k.startswith(first_layer_prefix))
     }
 
-    ntok = nearest_32(model_args.vision_chunk_ntok)
+    ntok = nearest_32(model_args.vision_chunk_ntok - 1)
```
Comment (cglagovichTT) [inline models/tt_transformers/tests/multimodal/test_llama_class_embedding.py:96]: we had an off-by-one error? what was it?

File: models/tt_transformers/tt/multimodal/llama_vision_encoder.py [hunk 2]
```diff
@@ -208,17 +208,12 @@ def forward(self, images, ar):
         ntok += 1
 
         # apply position embeddings
```
Comment (cglagovichTT) [inline models/tt_transformers/tt/multimodal/llama_vision_encoder.py:210]: this diff is beautiful

**Review by cglagovichTT — APPROVED** [review]

File: models/tt_transformers/tt/multimodal/llama_vision_encoder.py (hunk 2 above)
Comment (gwangTT) [reply models/tt_transformers/tt/multimodal/llama_vision_encoder.py:210]: 🤔

File: models/tt_transformers/tests/multimodal/test_llama_class_embedding.py (hunk 1 above)
Comment (gwangTT) [reply models/tt_transformers/tests/multimodal/test_llama_class_embedding.py:96]: `ntok` is used to construct the input tensor to `ClassEmbedding` which adds 1 to `ntok`. 

Another way to see this is that `model_args.vision_chunk_ntok` produces the `ntok` number after the `ClassEmbedding` is applied.

**Review by djordje-tt — APPROVED** [review]

**Comment by gwangTT** [comment]
Re-running tests before merging.

File: models/tt_transformers/scripts/repack_weights_90b.py [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+
+
+def chunk_key(key, chunk_size):
+    """
+    Return the chunk number that a key should go into
+    """
+    layer_id = layer_num(key)
+    assert layer_id >= 0, f"Unexpected key {key}"
+    chunk_id = layer_id // chunk_size
+    print(f"Key: {key} -> chunk_id: {chunk_id}")
```
Comment (bbradelTT) [inline models/tt_transformers/scripts/repack_weights_90b.py:37]: Would logger.info be better?

**Review by bbradelTT — APPROVED** [review]

File: models/tt_transformers/scripts/repack_weights_90b.py (hunk 3 above)
Comment (gwangTT) [reply models/tt_transformers/scripts/repack_weights_90b.py:37]: Yes, it would be better! Thanks, Borys!

**Review by afuller-TT — APPROVED** [review]

**Comment by gwangTT** [comment]
Thanks, @afuller-TT!
